src/core/handle.ml

```diff
@@ -263,6 +263,21 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
         | P_config_ident(id)      ->
             Sign.add_ident ss.signature id;
             out 3 "(conf) declared identifier [%s]\n" id; ss
+        | P_config_unif_rule(h)   ->
+            (* Approximately same processing as rules without SR checking. *)
+            let pur = (scope_rule ss h).elt in
+            let urule =
+              { lhs = pur.pr_lhs
+              ; rhs = Bindlib.unbox (Bindlib.bind_mvar pur.pr_vars pur.pr_rhs)
+              ; arity = List.length pur.pr_lhs
+              ; arities = pur.pr_arities
+              ; vars = pur.pr_vars }
+            in
+            Sign.add_rule Unif.Hint.sign Unif.Hint.atom urule;
```

Comment:
The rule should be rather added to the current signature here, right?

Reply:
Ah yes. It seems to me that it does not make a difference, but it keeps things tidier.

Reply:
Except that I must add the hint signature to the dependencies of all signatures.

Reply:
Yeah, there is no avoiding that I think.